### PR TITLE
dsumaroka\spring-boot-intro

### PR DESCRIPTION
**Pull request by dsumaroka**

### CODE REVIEW

File: src/main/java/mate/academy/springbootintro/model/Book.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+import lombok.NonNull;
+
+@Entity
+@Data
+@Table(name = "book")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/model/Book.java]:
```suggestion
@Getter
@Setter
@Table(name = "books")
```

File: src/main/java/mate/academy/springbootintro/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    private Book() {
+    }
+
+    public Book(@NonNull String title, @NonNull String author,
+                @NonNull String isbn, @NonNull BigDecimal price) {
+        this.title = title;
+        this.author = author;
+        this.isbn = isbn;
+        this.price = price;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/model/Book.java]: redundant constructors
in main use setters

File: src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.List;
+import mate.academy.springbootintro.model.Book;
+import mate.academy.springbootintro.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java:13]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert into DB" + book);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert into DB" + book);
```

File: src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all books from DB", e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all books from DB", e);
```

File: src/main/java/mate/academy/springbootintro/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package mate.academy.springbootintro.service.impl;
+
+import java.util.List;
+import mate.academy.springbootintro.model.Book;
+import mate.academy.springbootintro.repository.BookRepository;
+import mate.academy.springbootintro.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/service/impl/BookServiceImpl.java:10]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/springbootintro/service/impl/BookServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/service/impl/BookServiceImpl.java]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Comment by dsumaroka** [comment]
Ок

Ср, 11 сент. 2024 г. в 17:01, Olena Bruiako ***@***.***>:

File: src/main/java/mate/academy/springbootintro/model/Book.java [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.NonNull;
+
+@Entity
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/model/Book.java:14]:
```suggestion
@Entity
@Getter
@Setter
```

File: src/main/java/mate/academy/springbootintro/model/Book.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    private String author;
+    @NonNull
+    @Column(unique = true)
+    private String isbn;
+    @NonNull
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
+    public Book() {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/model/Book.java]: remove

File: src/main/java/mate/academy/springbootintro/model/Book.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public Long getId() {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/model/Book.java]: remove getters and setters

File: src/main/java/mate/academy/springbootintro/model/Book.java [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+    public String getCoverImage() {
+        return coverImage;
+    }
+
+    public void setCoverImage(String coverImage) {
+        this.coverImage = coverImage;
+    }
+
+    @Override
+    public String toString() {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/model/Book.java]: remove

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]
In next HW remove mate from package name

**Comment by dsumaroka** [comment]
Ок. Thanks

Чт, 12 сент. 2024 г. в 15:24, Olena Bruiako ***@***.***>:

**Pull request merged**